apps/prairielearn/assets/scripts/lib/tabulator.ts

```diff
@@ -0,0 +1,134 @@
+import { parseHTMLElement } from '@prairielearn/browser-utils';
+import { html } from '@prairielearn/html';
+import { v4 as uuid } from 'uuid';
+import {
+  Tabulator,
+  FormatModule,
+  EditModule,
+  FilterModule,
+  SortModule,
+  PageModule,
+  MutatorModule,
+  FrozenColumnsModule,
+  ResizeTableModule,
+  Options,
+} from 'tabulator-tables';
+
+Tabulator.registerModule([
+  FormatModule, // custom formatters
+  EditModule, // editors, also required for header filters
+  FilterModule, // column header filters
+  SortModule, // sorting by clicking on headers
+  PageModule, // pagination
+  MutatorModule, // calculate data derived from other columns
+  FrozenColumnsModule, // freeze first column
+  ResizeTableModule, // recalculate column widths on page resize
+]);
+
+type DefaultTabulatorOptions = Options & {
+  columnVisibilityDropdown?: Element | null;
+};
+
+export function defaultTabulator(
+  elementQueryString: string,
+  options: DefaultTabulatorOptions,
+): Tabulator {
+  const table: Tabulator = new Tabulator(elementQueryString, {
+    layout: 'fitData',
+    pagination: true,
```

Comment:
The pagination bar jumps when you get to the last page because the last page is typically shorter than the rest. Would it be possible to put the pagination bar above the table rather than below it? I would think that just above the table header would be ideal. This might also fix the slightly inconsistent pagination bar positioning when scrolling to the bottom of a long paginated page.

Reply:
I'm checking out the best way to do it, but https://github.com/olifolkerd/tabulator/issues/4488 is slowing me down a bit.

Reply:
See https://github.com/olifolkerd/tabulator/issues/4488#issuecomment-2227355942.